networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -306,4 +322,46 @@ def _asr_do_remove_floating_ip(self, ri, floating_ip,
                 (fixed_ip, floating_ip, vrf))
         self._edit_running_config(confstr,
                                   'REMOVE_STATIC_SRC_TRL_NO_VRF_MATCH')
-        self._asr_do_remove_secondary_ip(ri, floating_ip, ex_gw_port)
+
+    def _do_set_snat_pool(self, pool_name, pool_start,
+                          pool_end, pool_net, is_delete):
+        try:
+            if is_delete:
+                conf_str = asr1k_snippets.DELETE_NAT_POOL % (
+                    pool_name, pool_start, pool_end, pool_net)
+                # TODO(update so that hosting device name is passed down)
+                self._edit_running_config(conf_str, 'DELETE_NAT_POOL')
+
+            else:
+                conf_str = asr1k_snippets.CREATE_NAT_POOL % (
+                    pool_name, pool_start, pool_end, pool_net)
+                # TODO(update so that hosting device name is passed down)
+                self._edit_running_config(conf_str, 'CREATE_NAT_POOL')
+        except Exception as cse:
+            LOG.error(_LE("Temporary disable NAT_POOL exception handling: "
+                          "%s"), cse)
+
+    def _set_snat_pools_from_hosting_info(self, ri, gw_port, is_delete):
+        # TODO(tbachma ): unique naming for more than one pool
+        vrf_name = self._get_vrf_name(ri)
+        for subnet in gw_port['hosting_info'].get('snat_subnets', []):
+            net = netaddr.IPNetwork(subnet['cidr'])
+            pool_start = str(netaddr.IPAddress(net.first + 2))
+            pool_end = str(netaddr.IPAddress(net.first + 2))
+            pool_name = "%s_nat_pool" % (vrf_name)
+            self._do_set_snat_pool(pool_name, pool_start,
+                                   pool_end, str(net.netmask), is_delete)
+            secondary_ip = netaddr.IPAddress(net.value +
+                                             (net.hostmask.value - 1))
+            if not secondary_ip:
```

Comment:
I think this is leftover from a previous refactor. I'll remove, retest, and resubmit.